### PR TITLE
Mentions data driven

### PR DESCRIPTION
**Pull request by heiyu4585**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |   support options        |
| 🇨🇳 Chinese |       支持options    |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

File: components/mentions/index.tsx [hunk 1]
```diff
@@ -28,7 +28,7 @@ export interface OptionProps {
 }
 
 export interface DataDrivenOptionProps {
-  label?: React.ReactNode;
+  label?: string | React.ReactNode;
```
Comment (afc163) [inline components/mentions/index.tsx:31]: ReactNode 里面有 string，这里不需要改。

Comment (heiyu4585) [reply components/mentions/index.tsx]: get

File: components/mentions/index.en-US.md [hunk 2]
```diff
@@ -56,6 +68,7 @@ When you need to mention someone or something.
 | onResize | The callback function that is triggered when textarea resize | function({ width, height }) | - |  |
 | onSearch | Trigger when prefix hit | (text: string, prefix: string) => void | - |  |
 | onSelect | Trigger when user select the option | (option: OptionProps, prefix: string) => void | - |  |
+| options | Option Configuration | \[Options](#Option) | \[] | 4.25.0 |
```
Comment (zombieJ) [inline components/mentions/index.en-US.md:71]: 这个应该是 v5 的了

File: components/mentions/index.tsx [hunk 3]
```diff
@@ -109,17 +116,15 @@ const InternalMentions: React.ForwardRefRenderFunction<MentionsRef, MentionProps
...
-        <Option value="ANTD_SEARCHING" disabled>
-          <Spin size="small" />
-        </Option>
-      );
-    }
-
-    return children;
-  };
+  if (loading) {
+    options = [
```
Comment (zombieJ) [inline components/mentions/index.tsx:120]: 不要直接覆盖提供的参数，额外写一个 mergedOptions 来处理~

Comment (heiyu4585) [reply components/mentions/index.tsx]: done

File: components/mentions/index.en-US.md (hunk 2 above)
Comment (heiyu4585) [reply components/mentions/index.en-US.md]: done

**Review by zombieJ — APPROVED** [review]

File: components/mentions/index.zh-CN.md [hunk 4]
```diff
@@ -57,6 +69,7 @@ demo:
 | onResize | resize 回调 | function({ width, height }) | - |  |
 | onSearch | 搜索时触发 | (text: string, prefix: string) => void | - |  |
 | onSelect | 选择选项时触发 | (option: OptionProps, prefix: string) => void | - |  |
+| options | 选项配置 | \[Options](#Option) | [] | 5.1.0    |
```
Comment (hengkx) [inline components/mentions/index.zh-CN.md:72]:
```suggestion
| options | 选项配置 | [Options](#Option) | [] | 5.1.0    |
```

File: components/mentions/index.en-US.md [hunk 5]
```diff
@@ -56,6 +68,7 @@ When you need to mention someone or something.
 | onResize | The callback function that is triggered when textarea resize | function({ width, height }) | - |  |
 | onSearch | Trigger when prefix hit | (text: string, prefix: string) => void | - |  |
 | onSelect | Trigger when user select the option | (option: OptionProps, prefix: string) => void | - |  |
+| options | Option Configuration | \[Options](#Option) | \[] | 5.1.0       |
```
Comment (hengkx) [inline components/mentions/index.en-US.md:71]:
```suggestion
| options | Option Configuration | [Options](#Option) | \[] | 5.1.0       |
```

File: components/mentions/index.zh-CN.md (hunk 4 above)
Comment (heiyu4585) [reply components/mentions/index.zh-CN.md]: done

File: components/mentions/index.tsx [hunk 6]
```diff
@@ -78,6 +87,16 @@ const InternalMentions: React.ForwardRefRenderFunction<MentionsRef, MentionProps
   const [focused, setFocused] = React.useState(false);
   const innerRef = React.useRef<MentionsRef>();
   const mergedRef = composeRef(ref, innerRef);
+
+  // =================== Warning =====================
+  if (process.env.NODE_ENV !== 'production') {
+    warning(
```
Comment (zombieJ) [inline components/mentions/index.tsx:93]: 这个好像没写测试

Comment (heiyu4585) [reply components/mentions/index.tsx:93]: done

**Pull request merged**